clients/venice-client/src/test/java/com/linkedin/venice/fastclient/meta/utils/RequestBasedMetadataTestUtils.java

```diff
@@ -100,11 +113,28 @@ public static VeniceCompressor getZstdVeniceCompressor(String storeName) {
         .createVersionSpecificCompressorIfNotExist(CompressionStrategy.ZSTD_WITH_DICT, resourceName, DICTIONARY);
   }
 
+  public static RouterBackedSchemaReader getMockRouterBackedSchemaReader() {
+    RouterBackedSchemaReader metadataResponseSchemaReader = mock(RouterBackedSchemaReader.class);
+    int latestSchemaId = AvroProtocolDefinition.SERVER_METADATA_RESPONSE.getCurrentProtocolVersion();
+    when(metadataResponseSchemaReader.getLatestValueSchemaId()).thenReturn(latestSchemaId, latestSchemaId + 1);
+    doReturn(MetadataResponseRecord.SCHEMA$).when(metadataResponseSchemaReader).getValueSchema(latestSchemaId);
+    doReturn(MetadataResponseRecord.SCHEMA$).when(metadataResponseSchemaReader).getValueSchema(latestSchemaId + 1);
+    return metadataResponseSchemaReader;
+  }
+
   public static RequestBasedMetadata getMockMetaData(ClientConfig clientConfig, String storeName) {
+    return getMockMetaData(clientConfig, storeName, getMockRouterBackedSchemaReader());
+  }
+
+  public static RequestBasedMetadata getMockMetaData(
+      ClientConfig clientConfig,
+      String storeName,
+      RouterBackedSchemaReader routerBackedSchemaReader) {
     D2TransportClient d2TransportClient = RequestBasedMetadataTestUtils.getMockD2TransportClient(storeName);
     D2ServiceDiscovery d2ServiceDiscovery =
         RequestBasedMetadataTestUtils.getMockD2ServiceDiscovery(d2TransportClient, storeName);
-    RequestBasedMetadata requestBasedMetadata = new RequestBasedMetadata(clientConfig, d2TransportClient);
+    RequestBasedMetadata requestBasedMetadata =
+        new RequestBasedMetadata(clientConfig, d2TransportClient, routerBackedSchemaReader);
```

Comment:
I guess testing will get disrupted if we create the schema reader inside `RequestBasedMetadata`. We can add functions to set this schema reader and make it open only for tests by setting package private visibility